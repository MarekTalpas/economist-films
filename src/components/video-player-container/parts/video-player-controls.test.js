// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import VideoPlayerControls from './video-player-controls';

test('Video is playing + selected posstition is 0', () => {
  const tree : string = renderer.create(
    <VideoPlayerControls
       isVideoPlaying
       selectedPosition={ 0 }
       progress={0}
     />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Video is playing + selected posstition is 1', () => {
  const tree : string = renderer.create(
    <VideoPlayerControls
       isVideoPlaying
       selectedPosition={ 1 }
       progress={0}
     />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Video is playing + selected posstition is 2', () => {
  const tree : string = renderer.create(
    <VideoPlayerControls
       isVideoPlaying
       selectedPosition={ 2 }
       progress={0}
     />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Video is not playing ', () => {
  const tree : string = renderer.create(
    <VideoPlayerControls
       isVideoPlaying={ false }
       selectedPosition={ 0 }
       progress={0}
     />).toJSON();
  expect(tree).toMatchSnapshot();
});
