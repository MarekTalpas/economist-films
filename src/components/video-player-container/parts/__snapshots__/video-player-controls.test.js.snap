// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video is not playing  1`] = `
<div
  className="player-interface"
>
  <div
    className="player-interface__progress-bar-container"
  >
    <div
      className="player-interface__progress-bar"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
  <div
    className="player-interface__navigation-wrapper"
  >
    <button
      className="player-interface__navigation-control selected"
    >
      <img
        alt="left"
        className="player-interface__navigation-icons"
        id="1"
        src="RW.svg"
      />
    </button>
    <button
      className="player-interface__navigation-control"
    >
      <img
        alt="Play/Pause"
        className="player-interface__navigation-icons"
        id="2"
        src="Play.svg"
      />
    </button>
    <button
      className="player-interface__navigation-control"
    >
      <img
        alt="right"
        className="player-interface__navigation-icons"
        id="3"
        src="FF.svg"
      />
    </button>
  </div>
</div>
`;

exports[`Video is playing + selected postition is 0 1`] = `
<div
  className="player-interface"
>
  <div
    className="player-interface__progress-bar-container"
  >
    <div
      className="player-interface__progress-bar"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
  <div
    className="player-interface__navigation-wrapper"
  >
    <button
      className="player-interface__navigation-control selected"
    >
      <img
        alt="left"
        className="player-interface__navigation-icons"
        id="1"
        src="RW.svg"
      />
    </button>
    <button
      className="player-interface__navigation-control"
    >
      <img
        alt="Play/Pause"
        className="player-interface__navigation-icons"
        id="2"
        src="Pause.svg"
      />
    </button>
    <button
      className="player-interface__navigation-control"
    >
      <img
        alt="right"
        className="player-interface__navigation-icons"
        id="3"
        src="FF.svg"
      />
    </button>
  </div>
</div>
`;

exports[`Video is playing + selected postition is 1 1`] = `
<div
  className="player-interface"
>
  <div
    className="player-interface__progress-bar-container"
  >
    <div
      className="player-interface__progress-bar"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
  <div
    className="player-interface__navigation-wrapper"
  >
    <button
      className="player-interface__navigation-control"
    >
      <img
        alt="left"
        className="player-interface__navigation-icons"
        id="1"
        src="RW.svg"
      />
    </button>
    <button
      className="player-interface__navigation-control selected"
    >
      <img
        alt="Play/Pause"
        className="player-interface__navigation-icons"
        id="2"
        src="Pause.svg"
      />
    </button>
    <button
      className="player-interface__navigation-control"
    >
      <img
        alt="right"
        className="player-interface__navigation-icons"
        id="3"
        src="FF.svg"
      />
    </button>
  </div>
</div>
`;

exports[`Video is playing + selected postition is 2 1`] = `
<div
  className="player-interface"
>
  <div
    className="player-interface__progress-bar-container"
  >
    <div
      className="player-interface__progress-bar"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
  <div
    className="player-interface__navigation-wrapper"
  >
    <button
      className="player-interface__navigation-control"
    >
      <img
        alt="left"
        className="player-interface__navigation-icons"
        id="1"
        src="RW.svg"
      />
    </button>
    <button
      className="player-interface__navigation-control"
    >
      <img
        alt="Play/Pause"
        className="player-interface__navigation-icons"
        id="2"
        src="Pause.svg"
      />
    </button>
    <button
      className="player-interface__navigation-control selected"
    >
      <img
        alt="right"
        className="player-interface__navigation-icons"
        id="3"
        src="FF.svg"
      />
    </button>
  </div>
</div>
`;
